api-get-policy.go

```diff
@@ -35,7 +35,8 @@ func (c Client) GetBucketPolicy(bucketName, objectPrefix string) (bucketPolicy p
 		return policy.BucketPolicyNone, err
 	}
 	policyInfo, err := c.getBucketPolicy(bucketName)
-	if err != nil {
+	errResponse := ToErrorResponse(err)
+	if err != nil && errResponse.Code != "NoSuchBucketPolicy" {
```

Comment:
I think it is better if we return policy.BucketPolicyNone when we have NoSuchBucketPolicy without going to policy.GetPolicy() later.